backend/gadget.pm

```diff
@@ -0,0 +1,139 @@
+# Copyright © 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# this backend uses a KVM connector speaking VNC and external tools
+# for serial line and power cycling
+
+package backend::gadget;
+use strict;
+use base ('backend::baseclass');
+use bmwqemu qw(diag);
+use testapi qw(get_required_var get_var);
+use IPC::Run ();
+require IPC::System::Simple;
+use autodie qw(:all);
+
+sub new {
+    my $class = shift;
+    get_required_var('WORKER_HOSTNAME');
```

Comment:
upload_logs needs it for !qemu backends - and to avoid very late suprises, we check the existance of the variable early in these backends. you will see the same in svirt and ipmi